__tests__/app.test.js

```diff
@@ -93,6 +93,153 @@ describe('GET /api/articles', () => {
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSortedBy('topic', { descending: true });
+        });
+    });
+
+    it('200: responds with articles sorted by author', () => {
+      return request(app)
+        .get('/api/articles?sort_by=author')
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSortedBy('author', { descending: true });
+        });
+    });
+
+    it('200: responds with articles sorted by created_at', () => {
+      return request(app)
+        .get('/api/articles?sort_by=created_at')
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSortedBy('created_at', { descending: true });
+        });
+    });
+
+    it('200: responds with articles sorted by votes', () => {
+      return request(app)
+        .get('/api/articles?sort_by=votes')
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSortedBy('votes', { descending: true });
+        });
+    });
+
+    it('200: responds with articles sorted by article_img_url', () => {
+      return request(app)
+        .get('/api/articles?sort_by=article_img_url')
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSortedBy('article_img_url', { descending: true });
+        });
+    });
+
+    it('200: responds with articles sorted by comment_count', () => {
+      return request(app)
+        .get('/api/articles?sort_by=comment_count')
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSortedBy('comment_count', { descending: true });
+        });
+    });
+
+    it('400: responds with invalid sort_by query for an invalid sort_by query', () => {
+      return request(app)
+        .get('/api/articles?sort_by=invalid-column')
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.message).toBe('Invalid sort_by query');
+        });
+    });
+  });
+
+  describe('?order', () => {
+    it('200: responds with articles sorted in ascending order when order is asc', () => {
+      return request(app)
+        .get('/api/articles?order=asc')
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSortedBy('created_at', { ascending: true });
+        });
+    });
+
+    it('200: responds with articles sorted in descending order when order is desc', () => {
+      return request(app)
+        .get('/api/articles?order=desc')
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSortedBy('created_at', { descending: true });
+        });
+    });
+
+    it('400: responds with invalid order query for an invalid order query', () => {
+      return request(app)
+        .get('/api/articles?order=not-a-valid-order')
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.message).toBe('Invalid order query');
+        });
```

Comment:
we need a topic doesnt exist 404 :)

Reply:
There is a 404 test on lines 123-131.
Would you suggest ordering these by status code? At the moment they are just in the logical order I created them.